src/pyaro/timeseries/Filter.py

```diff
@@ -579,16 +579,15 @@ def init_kwargs(self):
         }
 
     def _index_from_include_exclude(self, times1, times2, includes, excludes):
-        idx = times1.astype("bool")
         if len(includes) == 0:
```

Comment:
Looks like `includes` is assumed to be an `Iterable`, in which case 
`if len(includes) == 0:` is equivalent to `if includes:`.

With this in mind, this section can be written as
``` python
        idx = np.repeat(not includes, len(times1))
        for start, end in includes:
            idx |= (np.datetime64(start) <= times1) & (times2 <= np.datetime64(end))

```

Reply:
While `idx = np.repeat(not includes, len(times1))` is written using very few characters, the boolean value of includes is not related to the boolean value of idx except for a chain of action, e.g. we could also end up with idx = np.repeat(includes, ...) if we would use the idx differently.

Rather than having to add long comments, please use rather more code.